test/rbs/signature_parsing_test.rb

```diff
@@ -1761,11 +1761,27 @@ def hello: (String _name) -> void
     RBS
   end
 
+  def test_underscore_alias_name
+    Parser.parse_signature(<<-RBS)
+class Hello
+  alias _foo _bar
+end
+    RBS
+  end
+
   def test_underscore_type_name
     assert_raises RBS::Parser::SyntaxError do
       Parser.parse_signature(<<-RBS)
 type _foo = Integer
       RBS
     end
   end
+
+  def test_underscore_qualified_name
+    assert_raises RBS::Parser::SyntaxError do
+      Parser.parse_signature(<<-RBS)
+type x = Foo::_bar
+      RBS
+    end
+  end
```

Comment:
This PR doesn't affect qualified_name, but I added the test for qualifed_name to clear that it doesn't accept tUNDERSCOREIDENT.